checker/src/main/java/org/checkerframework/checker/determinism/DeterminismVisitor.java

```diff
@@ -492,36 +500,45 @@ public Void visitMethod(MethodTree node, Void p) {
         boolean isPolyDownPresent = false;
         boolean isPolyUsePresent = false;
         boolean isPolyUpDetPresent = false;
-        for (AnnotationMirror atm : polyAnnotations) {
+        for (AnnotationMirror atm : paramAnnotations) {
```

Comment:
In the loop, the condition expression of every if statement is executed even if the first one is true.  Also, it would be more efficient to check if the annotation is `@PolyDet` with any argument, `AnnotationUtils#areSameByClass`, then get the argument once using `AnnotationUtils#getElementValue`.   Then all the if statements could be replaced with a switch.